src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLWebsocketMiddleware.fs

```diff
@@ -0,0 +1,366 @@
+        let rcv () = socket |> rcvMsgViaSocket serializerOptions
+
+        let sendOutput id (output : Output) =
+            match output.TryGetValue ("errors") with
+            | true, theValue ->
+                // The specification says: "This message terminates the operation and no further messages will be sent."
+                subscriptions
+                |> GraphQLSubscriptionsManagement.removeSubscription (id)
+                sendMsg (Error (id, unbox theValue))
+            | false, _ -> sendMsg (Next (id, output))
+
+        let sendSubscriptionResponseOutput id subscriptionResult =
+            match subscriptionResult with
+            | SubscriptionResult output -> output |> sendOutput id
+            | SubscriptionErrors (output, errors) ->
+                logger.LogWarning(
+                    "Subscription errors: {subscriptionerrors}",
+                    (String.Join ('\n', errors |> Seq.map (fun x -> $"- %s{x.Message}")))
+                )
+                Task.FromResult (())
+
+        let sendDeferredResponseOutput id deferredResult =
+            match deferredResult with
+            | DeferredResult (obj, path) ->
+                let output = obj :?> Dictionary<string, obj>
+                output |> sendOutput id
+            | DeferredErrors (obj, errors, _) ->
+                logger.LogWarning(
+                    "Deferred response errors: {deferrederrors}",
+                    (String.Join ('\n', errors |> Seq.map (fun x -> $"- %s{x.Message}")))
+                )
+                Task.FromResult (())
+
+        let sendDeferredResultDelayedBy (cancToken : CancellationToken) (ms : int) id deferredResult : Task = task {
+            do! Async.StartAsTask (Async.Sleep ms, cancellationToken = cancToken)
+            do! deferredResult |> sendDeferredResponseOutput id
+        }
+        let sendQueryOutputDelayedBy = sendDeferredResultDelayedBy cancellationToken
+
+        let applyPlanExecutionResult (id : SubscriptionId) (socket) (executionResult : GQLExecutionResult) : Task = task {
+            match executionResult with
+            | Stream observableOutput ->
+                (subscriptions, socket, observableOutput, serializerOptions)
+                |> addClientSubscription id sendSubscriptionResponseOutput
+            | Deferred (data, errors, observableOutput) ->
+                do! data |> sendOutput id
+                if errors.IsEmpty then
+                    (subscriptions, socket, observableOutput, serializerOptions)
+                    |> addClientSubscription id (sendQueryOutputDelayedBy 5000)
+                else
+                    ()
+            | Direct (data, _) -> do! data |> sendOutput id
+            | RequestError problemDetails ->
+                logger.LogWarning(
+                    "Request error: %s",
+                    (String.Join ('\n', problemDetails |> Seq.map (fun x -> $"- %s{x.Message}")))
```

Comment:
Why cannot we just log errors as is? They are records hence they already have ToString overridden.
```suggestion
                logger.LogWarning("Request errors:\n{errors}", problemDetails)
```
Is the intent to make errors prettier?